src/reaction_lammps_mupt/input_parser.py

```
@@ -0,0 +1,134 @@
+    def validate_no_duplicate_smiles(self, inputs: dict) -> None:
+        smiles_map = inputs.get("monomers", {})
+        seen = {}
+        for monomer_number, smi in smiles_map.items():
+            normalized = smi.strip()
+            if normalized in seen:
+                raise ValueError(
+                    f"Duplicate SMILES detected for monomers {seen[smi]} and {monomer_number}: {smi!r}"
+                )
+            seen[normalized] = monomer_number
+
+    def validate_inputs(self, inputs: dict) -> dict:
```

Ruff (0.14.14)

[error] Redefinition of unused `validate_inputs` from line 46: `validate_inputs` redefined here

(F811)